multicluster/controllers/multicluster/label_identity_export_controller_test.go

```diff
@@ -0,0 +1,355 @@
+			existResExp: &mcsv1alpha1.ResourceExportList{},
+			existResImp: &mcsv1alpha1.ResourceImportList{
+				Items: []mcsv1alpha1.ResourceImport{*labelIdentityResImp},
+			},
+			resExpNamespacedName:     resExpNamespacedName,
+			expNormalizedLabel:       "",
+			originalLabelsToClusters: map[string]sets.String{labelHash: sets.NewString(localClusterID)},
+			originalClusterToLabels:  map[string]sets.String{localClusterID: sets.NewString(labelHash)},
+			expLabelsToClusters:      map[string]sets.String{},
+			expClusterToLabels:       map[string]sets.String{localClusterID: sets.NewString()},
+			expLabelResImpDeleted:    true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithLists(tt.existResExp, tt.existResImp).Build()
+			r := NewLabelIdentityExportReconciler(fakeClient, scheme, leaderNamespace)
+			if len(tt.originalLabelsToClusters) > 0 {
+				r.clusterToLabels = tt.originalClusterToLabels
+				r.labelsToClusters = tt.originalLabelsToClusters
+			}
+			go r.Run(stopCh)
+			resExpReq := ctrl.Request{NamespacedName: tt.resExpNamespacedName}
+			if _, err := r.Reconcile(ctx, resExpReq); err != nil {
+				t.Errorf("LabelIdentityExport Reconciler got error during reconciling. error = %v", err)
+			}
+
+			if !reflect.DeepEqual(r.labelsToClusters, tt.expLabelsToClusters) {
+				t.Errorf("LabelIdentityExport Reconciler operated labelsToClusters incorrectly. Exp: %s, Act: %s", tt.expLabelsToClusters, r.labelsToClusters)
+			}
+			if !reflect.DeepEqual(r.clusterToLabels, tt.expClusterToLabels) {
+				t.Errorf("LabelIdentityExport Reconciler operated clusterToLabels incorrectly. Exp: %s, Act: %s", tt.expClusterToLabels, r.clusterToLabels)
+			}
+
+			time.Sleep(100 * time.Millisecond)
+			actLabelIdentityResImp := &mcsv1alpha1.ResourceImport{}
+			lastIdx := strings.LastIndex(tt.resExpNamespacedName.Name, "-")
+			parsedLabelHash := tt.resExpNamespacedName.Name[lastIdx+1:]
+			err := fakeClient.Get(ctx, types.NamespacedName{Namespace: "default", Name: parsedLabelHash}, actLabelIdentityResImp)
+			assert.Equalf(t, tt.expLabelResImpDeleted, apierrors.IsNotFound(err), "Unexpected error status when getting LabelIdentity kind of ResourceImport: %v", err)
+			if tt.expNormalizedLabel != "" {
+				actualLabel := actLabelIdentityResImp.Spec.LabelIdentity.Label
+				assert.Equalf(t, tt.expNormalizedLabel, actualLabel, "Unexpected normalized label in ResourceImport. Exp: %s, Act: %s", tt.expNormalizedLabel, actualLabel)
+			}
+		})
+	}
+}
+
+func TestIDAllocatorBasic(t *testing.T) {
+	testSteps := []struct {
+		op                        string
+		idExpectedOrReleasing     uint32
+		expectedAvailableReuseIDs int
+		expectErr                 bool
+	}{
+		{
+			"allocate",
+			1,
+			0,
+			false,
+		},
+		{
+			"allocate", 2,
+			0,
+			false,
+		},
+		{
+			"allocate", 3,
+			0,
+			false,
+		},
```

Comment:
```suggestion
		{
			"allocate",
			3,
			0,
			false,
		},
```
Please also fix other occurrences.

Reply:
Fixed